src/axom/core/Array.hpp

```diff
@@ -550,121 +830,303 @@ Array< T >::~Array()
   m_data = nullptr;
 }
 
+//------------------------------------------------------------------------------
+template< typename T >
+inline void Array< T >::fill( const T& value )
+{
+  for (IndexType i = 0 ; i < m_num_elements ; i++)
+  {
+    m_data[ i ] = value;
+  }
+}
+
+//------------------------------------------------------------------------------
+template< typename T >
+inline void Array< T >::push_back( const T& value )
+{
+  emplace_back(value);
+}
 
 //------------------------------------------------------------------------------
 template< typename T >
-inline void Array< T >::append( const T& value )
+inline void Array< T >::push_back( T&& value )
 {
-  assert( m_num_components == 1 );
+  emplace_back(std::move(value));
+}
 
-  IndexType new_size = m_num_tuples + 1;
-  if ( new_size > m_capacity )
+//------------------------------------------------------------------------------
+template< typename T >
+inline void Array< T >::set( const T* elements, IndexType n, IndexType pos )
+{
+  assert( elements != nullptr );
+  assert( pos >= 0 );
+  assert( pos + n <= m_num_elements );
+
+  for ( IndexType i = 0 ; i < n ; ++i )
```

Comment:
Ditto. As with the ``fill()`` we should probably implement this with an ``axom::for_all`` kernel launch and add a second, optional template argument to specify the execution space.